Fixed the blank spaces appearing on the exercise portal and made the page responsive.

KaustubhK98 opened the pull request:
## Description

The exercise module was using a "Grid Layout" to handle the exercise card actions. I replaced the "Grid Layout" with a "Recycler View". Now, the exercise cards will be dynamically added or removed from the adapter and after every action the page will be automatically refreshed. This issue has fixed the bug, which lead to blank spaces on the page and made the UI more responsive. Recycler View is a future proof solution over Grid Layout due to its ability to dynamically handle card actions and scalability. 

### Todos

- [ ] Tested and working locally
- [ ] Code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] Code changes documented
- [ ] Requested review from >= 2 devs on the team (one frontend and one backend recommended)

### How to test

1. Login to app
2. Go to exercise module page
3. Select any module and click on "mark as complete" or "save for later" button.
4. The card should be gone and there shouldn't be any blank space on the page.
5. The UI is dynamic now and is updated automatically after any changes are made to the exercise cards.

### Screenshots and/or Gifs

Uploading Screen Recording 2025-08-10 at 9.24.52 pm.mov…


### Associated MS Planner Tasks

- Exercise Page (https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/planner.v1.1c7aff73-bd5d-4eaf-844f-a0f0914d35b0_p_obnY9eRGNEGXqjBUOhDWKcgAFWt1?tenantId=d02378ec-1688-46d5-8540-1c28b5f470f6&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fv1%2Fplan%2FobnY9eRGNEGXqjBUOhDWKcgAFWt1%2Fview%2Fboard%2Ftask%2FGEU8WYnCaEWDe2v9jRFttcgAHubp%22%2C%22channelId%22%3A%2219%3A02e7479536e247c5a85093cf6dcaca4c%40thread.tacv2%22%7D)


### Known Issues

There aren't any known issues arising due to this change.

KaustubhK98 committed:
fixed the ui issue leading to blank spaces after marking exercise modules as complete or save as later -Kaustubh

AryanSharma1017 commented:
Hey Kaustubh, just check for the build once again by running ./gradlew spotlessapply and then check for the ./gradlew build and then see, I think then all these errors should go. 
if not then please have a talk with me and we will see.

KaustubhK98 committed:
Fixed the gradle issue which led to unsuccessful merging of pull request -Kaustubh

KaustubhK98 commented:
Hi Aryan,

Spotless build wasn't an issue. There was a Gradle version mismatch, I have fixed the problem on my end and raised another PR. Can you please let me know if you are able to see the new PR. Please try to merge the code now, I am certain that it will get merged this time. Please let me know if there are any problems while merging the new PR.

Thanks!

KaustubhK98 committed:
Night mode removed -Kaustubh

KaustubhK98 committed:
Fixed gradle spotless related issues -Kaustubh

KaustubhK98 committed:
Update ApiClient base URL

KaustubhK98 committed:
Added a functionality to assign a doctor to a patient and send a request to the doctor. Also, added a check to prevent sending duplicate requests. Added a search bar to search for doctors from the list

KaustubhK98 commented:
**1. Added functionality to assign doctor to a patient from the caretaker's profile:** 

- A caretaker can now send a request to a doctor before actually assigning a patient to them. 
- A caretaker can directly assign a patient to the doctor 
- A caretaker can Un-assign or cancel a request sent to a doctor. 
- The app doesn’t allow duplicate requests 
- Doctors can be searched using the search bar. 

**2. Safely removed the night mode and it's files from the app.** 

**3. Added a function to view the list of patients assigned to a doctor. This function is partially implemented and needs the doctor's portal to work properly.** 


**Known Issues**
There aren't any known issues arising due to this change.
